firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,292 @@
+/*********************************************************************************
+* @file           : mid_dabs.c
+* @brief          : Implementation of MID_DABS
+***********************************************************************************/
+
+/**********************************************************************************/
+/*                  Include common and project definition header                  */
+/**********************************************************************************/
+//#include "hal_adc.h"
+#include "hal_sts.h"
+#include "hal_gpio.h"
+#include "hal_tmr.h"
+#include "epc_conf.h"
+
```

Comment:
hal_gpio will be needed here

Reply:
This is something to review as I think the only needed are hal_gpio, hal_adc,  hal_sts and epc_conf